<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>638</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
    <horstretch>1</horstretch>
    <verstretch>1</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>100</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="gridWidget_5" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>1001</width>
      <height>521</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>2</horstretch>
      <verstretch>2</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>150</width>
      <height>100</height>
     </size>
    </property>
    <property name="sizeIncrement">
     <size>
      <width>2</width>
      <height>2</height>
     </size>
    </property>
    <property name="contextMenuPolicy">
     <enum>Qt::NoContextMenu</enum>
    </property>
    <layout class="QGridLayout" name="gridLayout_5" rowstretch="0,0" columnstretch="0,0">
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <property name="verticalSpacing">
      <number>10</number>
     </property>
     <item row="1" column="0">
      <layout class="QVBoxLayout" name="verticalLayout_5" stretch="0,0,0,0">
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1,1,1">
         <item>
          <widget class="QPushButton" name="fileInputButton">
           <property name="text">
            <string>Загрузить объект</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Изменить масштаб</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <property name="sizeConstraint">
            <enum>QLayout::SetMaximumSize</enum>
           </property>
           <item row="3" column="0">
            <widget class="QLabel" name="label">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>x</string>
             </property>
             <property name="margin">
              <number>4</number>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QPushButton" name="scaleToLessButton">
             <property name="text">
              <string>меньше</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="3">
            <widget class="QSlider" name="scaleSlider">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>z</string>
             </property>
             <property name="margin">
              <number>4</number>
             </property>
            </widget>
           </item>
           <item row="5" column="1" colspan="2">
            <widget class="QDoubleSpinBox" name="moveZDoubleSpinBox"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>y</string>
             </property>
             <property name="margin">
              <number>4</number>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="scaleToMoreButton">
             <property name="text">
              <string>больше</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1" colspan="2">
            <widget class="QDoubleSpinBox" name="moveXDoubleSpinBox">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="4" column="1" colspan="2">
            <widget class="QDoubleSpinBox" name="moveYDoubleSpinBox"/>
           </item>
           <item row="2" column="0" colspan="3">
            <widget class="QLabel" name="label_15">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Перемещение</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout">
         <property name="verticalSpacing">
          <number>3</number>
         </property>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="spinXdoubleSpinBox"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>x</string>
           </property>
           <property name="margin">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="spinYdoubleSpinBox"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>y</string>
           </property>
           <property name="margin">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>z</string>
           </property>
           <property name="margin">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="spinZdoubleSpinBox"/>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>Поворот</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="2" column="0">
          <widget class="QPushButton" name="linesThiknessLessButton">
           <property name="text">
            <string>меньше</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0" colspan="2">
          <widget class="QSlider" name="linesThiknessSlider">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="linesThiknessMoreButton">
           <property name="text">
            <string>больше</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QLabel" name="label_7">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Толщина ребер</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="sizeConstraint">
          <enum>QLayout::SetMaximumSize</enum>
         </property>
         <item row="1" column="0">
          <widget class="QPushButton" name="verticesThiknessLessButton">
           <property name="text">
            <string>-</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="verticesThiknessMoreButton">
           <property name="text">
            <string>+</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="label_3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Толщина вершин</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QSlider" name="verticesThiknessSlider">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="fileNameOutup">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="glView" name="GLWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
         <horstretch>100</horstretch>
         <verstretch>100</verstretch>
        </sizepolicy>
       </property>
       <property name="sizeIncrement">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="scrinshotButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>550</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Бум</string>
    </property>
   </widget>
   <widget class="QPushButton" name="gifButton">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>550</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Бам</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Сделать скриншот</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>530</y>
      <width>172</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Сделать GIF анимацию</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>530</y>
      <width>351</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Количество вершин</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_output_v">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>530</y>
      <width>172</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>530</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Количество ребер</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_output_l">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>530</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMenu">
    <property name="title">
     <string>Menu</string>
    </property>
    <widget class="QMenu" name="menuSet_projection_type">
     <property name="title">
      <string>Set projection type</string>
     </property>
     <addaction name="action1"/>
     <addaction name="action2"/>
    </widget>
    <widget class="QMenu" name="menuEdges">
     <property name="title">
      <string>Edges</string>
     </property>
     <addaction name="actionSolid"/>
     <addaction name="actionDashed"/>
     <addaction name="separator"/>
     <addaction name="actionSet_color_lines"/>
    </widget>
    <widget class="QMenu" name="menuVertices">
     <property name="title">
      <string>Vertices</string>
     </property>
     <addaction name="action_None"/>
     <addaction name="actionCircle"/>
     <addaction name="actionSquere"/>
     <addaction name="separator"/>
     <addaction name="actionSet_color_vertices"/>
    </widget>
    <widget class="QMenu" name="menuScrinshot_type">
     <property name="title">
      <string>Scrinshot type</string>
     </property>
     <addaction name="action_bmp"/>
     <addaction name="action_jpeg"/>
    </widget>
    <addaction name="actionSet_background_color"/>
    <addaction name="menuSet_projection_type"/>
    <addaction name="menuEdges"/>
    <addaction name="menuVertices"/>
    <addaction name="menuScrinshot_type"/>
   </widget>
   <addaction name="menuMenu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSet_background_color">
   <property name="text">
    <string>Set background color</string>
   </property>
  </action>
  <action name="action1">
   <property name="text">
    <string>Parallel</string>
   </property>
  </action>
  <action name="action2">
   <property name="text">
    <string>Central</string>
   </property>
  </action>
  <action name="actionSolid">
   <property name="text">
    <string>Solid</string>
   </property>
  </action>
  <action name="actionDashed">
   <property name="text">
    <string>Dashed</string>
   </property>
  </action>
  <action name="actionSet_color_lines">
   <property name="text">
    <string>Set color</string>
   </property>
  </action>
  <action name="actionSet_Thikness">
   <property name="text">
    <string>Set thikness</string>
   </property>
  </action>
  <action name="action_None">
   <property name="text">
    <string>None</string>
   </property>
  </action>
  <action name="actionCircle">
   <property name="text">
    <string>Circle</string>
   </property>
  </action>
  <action name="actionSquere">
   <property name="text">
    <string>Square</string>
   </property>
  </action>
  <action name="actionSet_color_vertices">
   <property name="text">
    <string>Set color</string>
   </property>
  </action>
  <action name="actionSet_size">
   <property name="text">
    <string>Set size</string>
   </property>
  </action>
  <action name="action_bmp">
   <property name="text">
    <string>.bmp</string>
   </property>
  </action>
  <action name="action_jpeg">
   <property name="text">
    <string>.jpeg</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>glView</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">glview.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
